resolver/custom_dns_resolver.go

```diff
@@ -120,26 +144,113 @@ func (r *CustomDNSResolver) processRequest(request *model.Request) *model.Respon
 		}
 	}
 
-	return nil
+	logger.WithField("next_resolver", Name(r.next)).Trace("go to next resolver")
+
+	forwardResponse, err := r.next.Resolve(ctx, request)
+	if err != nil {
+		return nil, resolvedCnames, err
+	}
+
+	return forwardResponse, resolvedCnames, nil
+}
+
+func (r *CustomDNSResolver) processRequest(ctx context.Context, request *model.Request) (*model.Response, error) {
+	response, _, err := r.processRequestTrackingCnames(ctx, request, make([]string, 0))
+	return response, err
+}
+
+func (r *CustomDNSResolver) processDNSEntry(
+	ctx context.Context,
+	request *model.Request,
+	resolvedCnames []string,
+	question dns.Question,
+	entry dns.RR,
+) ([]dns.RR, error) {
+	switch v := entry.(type) {
+	case *dns.A:
+		return r.processIP(v.A, question)
+	case *dns.AAAA:
+		return r.processIP(v.AAAA, question)
+	case *dns.CNAME:
+		return r.processCNAME(ctx, request, *v, resolvedCnames, question)
+	}
+
+	return nil, fmt.Errorf("unsupported type %T", entry)
```

Comment:
Trying to make what it means obvious even without the context of the code:
```suggestion
	return nil, fmt.Errorf("unsupported customDNS RR type %T", entry)
```